huaweicloud/services/css/resource_huaweicloud_css_logstash_configuration.go

```diff
@@ -0,0 +1,300 @@
+		Setting: &model.Setting{
+			QueueType:             d.Get("setting.0.queue_type").(string),
+			Workers:               utils.Int32IgnoreEmpty(int32(d.Get("setting.0.workers").(int))),
+			BatchSize:             utils.Int32IgnoreEmpty(int32(d.Get("setting.0.batch_size").(int))),
+			BatchDelayMs:          utils.Int32IgnoreEmpty(int32(d.Get("setting.0.batch_delay_ms").(int))),
+			QueueCheckPointWrites: utils.Int32IgnoreEmpty(int32(d.Get("setting.0.queue_check_point_writes").(int))),
+			QueueMaxBytesMb:       utils.Int32IgnoreEmpty(int32(d.Get("setting.0.queue_max_bytes_mb").(int))),
+		},
+	}
+
+	sensitiveWordsRaw := d.Get("sensitive_words").([]interface{})
+	if len(sensitiveWordsRaw) > 0 {
+		createOpts.SensitiveWords = utils.ExpandToStringListPointer(sensitiveWordsRaw)
+	}
+
+	return createOpts
+}
+
+func resourceLogstashConfigurationRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	region := conf.GetRegion(d)
+	cssV1Client, err := conf.HcCssV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating CSS V1 client: %s", err)
+	}
+
+	var mErr *multierror.Error
+
+	logstashConfDetail, err := cssV1Client.ShowGetConfDetail(&model.ShowGetConfDetailRequest{
+		ClusterId: d.Get("cluster_id").(string),
+		Name:      d.Get("name").(string),
+	})
+	if err != nil {
+		common.CheckDeletedDiag(d, err, "CSS logstash cluster configuration")
+	}
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("name", logstashConfDetail.Name),
+		d.Set("conf_content", logstashConfDetail.ConfContent),
+		d.Set("setting", flattenLogstashConfSetting(logstashConfDetail)),
+		d.Set("status", logstashConfDetail.Status),
+		d.Set("updated_at", logstashConfDetail.UpdateAt),
+	)
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func flattenLogstashConfSetting(logstashConfDetail *model.ShowGetConfDetailResponse) []map[string]interface{} {
+	result := make([]map[string]interface{}, 0)
+	result = append(result, map[string]interface{}{
+		"workers":                  logstashConfDetail.Setting.Workers,
+		"batch_size":               logstashConfDetail.Setting.BatchSize,
+		"batch_delay_ms":           logstashConfDetail.Setting.BatchDelayMs,
+		"queue_type":               logstashConfDetail.Setting.QueueType,
+		"queue_check_point_writes": logstashConfDetail.Setting.QueueCheckPointWrites,
+		"queue_max_bytes_mb":       logstashConfDetail.Setting.QueueMaxBytesMb,
+	})
+
+	return result
+}
+
+func resourceLogstashConfigurationUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	region := conf.GetRegion(d)
+	cssV1Client, err := conf.HcCssV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating CSS V1 client: %s", err)
+	}
+
+	updateLogstashConfChanges := []string{
+		"conf_content",
+		"setting",
+		"sensitive_words",
+	}
+
+	if d.HasChanges(updateLogstashConfChanges...) {
```

Comment:
if there is only one API used in the update func, we don't need to check the changes
```suggestion
```